localstack-core/localstack/services/lambda_/provider.py

```diff
@@ -1866,6 +1869,39 @@ def update_alias(
     # =======================================
     # ======= EVENT SOURCE MAPPINGS =========
     # =======================================
+    def check_service_resource_exists(self, service, resource_arn):
+        arn = parse_arn(resource_arn)
+        source_client = get_internal_client(resource_arn)
+        if service in ["sqs", "sqs-fifo"]:
+            try:
+                source_client.get_queue_url(QueueName=arn["resource"])
+            except ClientError as e:
+                if e.response["Error"]["QueryErrorCode"] == "QueueDoesNotExist":
+                    raise InvalidParameterValueException(
+                        "Error occurred while ReceiveMessage. SQS Error Code: AWS.SimpleQueueService.NonExistentQueue. SQS Error Message: The specified queue does not exist.",
+                        Type="User",
+                    )
+                raise e
+        elif service in ["kinesis"]:
+            try:
+                source_client.describe_stream(StreamARN=resource_arn)
+            except ClientError as e:
+                if e.response["Error"]["Code"] == "ResourceNotFoundException":
+                    raise InvalidParameterValueException(
+                        f"Stream not found: {resource_arn}",
+                        Type="User",
+                    )
+                raise e
+        elif service in ["dynamodb"]:
+            try:
+                source_client.describe_stream(StreamArn=resource_arn)
```

Comment:
Same here, this is the right operation! https://docs.aws.amazon.com/aws-managed-policy/latest/reference/AWSLambdaDynamoDBExecutionRole.html